tests/compile-fail/array_indexing.rs

```diff
@@ -8,6 +9,20 @@ fn main() {
     let x = [1,2,3,4];
     x[0];
     x[3];
-    x[4]; //~ERROR: const index-expr is out of bounds
-    x[1 << 3]; //~ERROR: const index-expr is out of bounds
+    x[4]; //~ERROR: const index is out of bounds
+          //^ERROR: indexing/slicing may panic
+    x[1 << 3]; //~ERROR: const index is out of bounds
+               //^ERROR: indexing/slicing may panic
+    &x[1..5]; //~ERROR: range is out of bounds
+              //^ERROR: indexing/slicing may panic
+    &x[..];
+    &x[1..];
+    &x[..4];
+    &x[..5]; //~ERROR: range is out of bounds
+             //^ERROR: indexing/slicing may panic
```

Comment:
Can you add some tests with the new inclusive range syntax here? (`...`)
I bet someone will be one off someday.